static/app/components/forms/fields/sentryMemberTeamSelectorField.tsx

```diff
@@ -77,11 +85,36 @@ function SentryMemberTeamSelectorField({
     loadMore: loadMoreTeams,
   } = useTeams({provideUserTeams: true});
 
-  const teamOptions = teams?.map(team => ({
-    value: `team:${team.id}`,
-    label: `#${team.slug}`,
-    leadingItems: <Avatar team={team} size={avatarSize} />,
-  }));
+  // TODO(davidenwang): Fix the team type here to avoid this type cast
+  const teamOptions = (teams as DetailedTeam[])
+    .map(team => {
+      const isDisabledTeam =
+        memberOfProjectSlug &&
+        !defined(team.projects.find(({slug}) => memberOfProjectSlug === slug));
+      return {
+        value: `team:${team.id}`,
+        leadingItems: <Avatar team={team} size={avatarSize} />,
+        ...(isDisabledTeam
+          ? {
+              disabled: true,
+              label: (
+                <Tooltip
+                  position="left"
+                  title={t('%s is not a member of the selected project', `#${team.slug}`)}
+                >
+                  #{team.slug}
+                </Tooltip>
+              ),
+            }
+          : {
+              disabled: false,
```

Comment:
probably can just leave this out

Reply:
mmm i had it in to make the sort easier to write since it compares on `disabled`, but we can move off this and the sort based on the other suggestion 🚀